nemo/export/trt_llm/nemo_utils.py

```diff
@@ -0,0 +1,327 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import ast
+import configparser
+import copy
+import csv
+import datetime
+import logging
+import os
+import shutil
+import sys
+import tempfile
+import typing
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import numpy as np
+import tensorrt_llm
+from tensorrt_llm import str_dtype_to_trt
+from transformers import AutoTokenizer, GPT2Config, LlamaConfig, PretrainedConfig, PreTrainedTokenizer
+
+from .model_config import (
+    LAYERNORM_DEFAULT,
+    LAYERNORM_RMS,
+    LINEAR_COLUMN,
+    DecoderLayerConfig,
+    EmbeddingConfig,
+    LayernormConfig,
+    LinearConfig,
+    ModelConfig,
+)
+from .nemo.nemo import UnpackedNemoCheckpointDir, unpack_nemo_ckpt
+from .nemo.nemo_ckpt_convert import build_tokenizer, convert_dist_checkpoint, convert_nemo_model
+from .tensor_utils import get_tensor_from_dict, get_tensor_parallel_group, split
+
+LOGGER = logging.getLogger("NeMo")
+
+
+def _nemo_decode(
```

Comment:
Please avoid genetic nams as `nemo_xyz` to mean `nemo_llm_xyz`. Please be explicit in all places you name things.

Reply:
Done.